requests/models.py

```diff
@@ -820,6 +827,9 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             for line in lines:
                 yield line
 
+            # check if the current chunk ends with '\r'
+            last_chunk_ends_with_cr = chunk.endswith('\r' if decode_unicode else b'\r')
```

Comment:
Let's hoist the result of these two `else` calls to the top of the function, rather than check it twice each time around the loop:

```python
carriage_return = u'\r' if decode_unicode else b'\r'
line_feed = u'\n' if decode_unicode else b'\r'
```